### ISSUE TITLE
Send from useCacheSend sends trasactions twice.

### ISSUE THREAD

**Issue by SalahAdDin** [trufflesuite/drizzle#101]
Hi guys, I'm following the old documentation about [hooks](https://github.com/trufflesuite/drizzle/blob/develop/packages/react-plugin/README.md#usecachesend) and I'm getting a problem at sending transactions: every transaction is sent twice, why?

The next is the current core:
 ```
 <form
             onSubmit={React.useCallback(
               ({
                 to = drizzleState.clientAccount,
                 value = toHex(parseAmountToToken(amountToSend, tokenDecimals)),
               }) => send(to, value),
               [amountToSend]
             )}
           >
             <label>Amount:</label>
             <input
               type="number"
               name="value"
               onChange={(e) => setAmountToSend(e.target.value)}
             />
             <button type="submit">Send</button>
           </form>
 ```
 
 Also `TXObjects` is ever undefined:  `<div>{TXObjects?.map((t) => t.status)}</div>`.
 
I'm seeing the [`send` function](https://github.com/trufflesuite/drizzle/blob/develop/packages/react-plugin/src/hooks/create-use-cache-send.js#L16) is calling the contract method twice for each transaction:
![Screen Capture_select-area_20200903141705](https://user-images.githubusercontent.com/5034215/92113880-aa1e3480-edf8-11ea-88ef-925cf40d3155.png)

But I don't understand why it is happening.

Now, if i decide use `cacheSend` directly, i don't get duplicated transactions:
```
  const sendTokens = (e) => {
    e.preventDefault();
    contract.cacheSend(
      drizzleState.clientAccount,
      toHex(parseAmountToToken(amountToSend, tokenDecimals))
    );
  };
```

 Can anyone guide me? Is this a bug?

**Comment by kombos** [comment]
it could probably be because since you are using the dependancy array [amountToSend], which means the callback will be fired whenever the value of amountToSend changes. Now, during the submit action, the amountToSend changes twice. First its value is whatever is set by user, then the next value becomes "" as the submit action erases the value. so the both times the callback is being fired. not sure but this could be the problem.

**Comment by SalahAdDin** [comment]
@kombos No, even when we use `send` with memoization i still get the problem:
```jsx
  const sendTokens = (e) => {
    e.preventDefault();
    send(
      drizzleState.clientAccount,
      toHex(parseAmountToToken(amountToSend, tokenDecimals))
    );
  };
```

@cds-amal The problem is [here](https://github.com/trufflesuite/drizzle/blob/develop/packages/react-plugin/src/hooks/create-use-cache-send.js#L16): if we remove the input variable and we left just the array, it won't repeat the transaction one more time.

**Comment by cds-amal** [comment]
Hi @SalahAdDin,

Thanks for the investigation. Can you provide a repo with the problem please? I'm not quite following your explanation.

**Comment by SalahAdDin** [comment]
> Hi @SalahAdDin,
> 
> Thanks for the investigation. Can you provide a repo with the problem please? I'm not quite following your explanation.

@cds-amal I can't, but I can explain the problem.

The main goal for us is having a form from you add a token amount and you do click on send to transfer that amount of tokens:
![image](https://user-images.githubusercontent.com/5034215/92299742-d7382780-ef5d-11ea-9aa6-f9ae69c198de.png)

Following the [documentation](https://github.com/trufflesuite/drizzle/tree/develop/packages/react-plugin#usecachesend) we tried this first:
```
      <TransferForm
        onSubmit={
          // Use memoization to avoid unnecessary rerenders.
          useCallback(({ to, value }) => send(to, value), [])
        }
      />
 ```

But it has the effect to send transactions twice:
![Screen Capture_select-area_20200905001225](https://user-images.githubusercontent.com/5034215/92299901-3cd8e380-ef5f-11ea-9b47-30e30d7d575e.png)
![Screen Capture_select-area_20200905001245](https://user-images.githubusercontent.com/5034215/92299903-41050100-ef5f-11ea-9c9a-ba4e46bbc9f1.png)

Therefore, we tried another approach using barebone `cacheSend`:
```
  const sendTokens = (e) => {
    e.preventDefault();
    contract.cacheSend(
      drizzleState.clientAccount,
      toHex(parseAmountToToken(amountToSend, tokenDecimals))
    );
  };
```
And we didn't have duplicated transactions.

Later, as we want to track the transactions, we made a copy of the hook functionality and we got the same: duplicated transactions.
```
  const sendTokens = (e) => {
    e.preventDefault();
    setStackIDs(stackIDs => [
      ...stackIDs,
    contract.cacheSend(
      account,
      toHex(parseAmountToToken(amountToSend, tokenDecimals))
      ),
    ]);
  };
```

Removing `stackIDs => ` from the set `setStackIDs` avoids transaction's duplicating, hence we have now the same function without that input:
```
const sendTokens = (e) => {
    e.preventDefault();
    setStackIDs([
      ...stackIDs,
      contract.cacheSend(
        "0x3cCe2F1442A0E55616B78369F280ae22D6d9a6Fc",
        account,
        toHex(parseAmountToToken(amountToSend, tokenDecimals))
      ),
    ]);
  };
```
